Alignment/MillePedeAlignmentAlgorithm/plugins/MillePedeDQMModule.cc

```diff
@@ -45,14 +46,61 @@ void MillePedeDQMModule ::bookHistograms(DQMStore::IBooker& booker) {
   edm::LogInfo("MillePedeDQMModule") << "Booking histograms";
 
   booker.cd();
-  booker.setCurrentFolder("AlCaReco/SiPixelAli/");
-
-  h_xPos = booker.book1D("Xpos", "Alignment fit #DeltaX;;#mum", 36, 0., 36.);
-  h_xRot = booker.book1D("Xrot", "Alignment fit #Delta#theta_{X};;#murad", 36, 0., 36.);
-  h_yPos = booker.book1D("Ypos", "Alignment fit #DeltaY;;#mum", 36, 0., 36.);
-  h_yRot = booker.book1D("Yrot", "Alignment fit #Delta#theta_{Y};;#murad", 36, 0., 36.);
-  h_zPos = booker.book1D("Zpos", "Alignment fit #DeltaZ;;#mum", 36, 0., 36.);
-  h_zRot = booker.book1D("Zrot", "Alignment fit #Delta#theta_{Z};;#murad", 36, 0., 36.);
+  if (!isHG_) {
+    booker.setCurrentFolder("AlCaReco/SiPixelAli/");
+    h_xPos = booker.book1D("Xpos", "Alignment fit #DeltaX;;#mum", 36, 0., 36.);
+    h_xRot = booker.book1D("Xrot", "Alignment fit #Delta#theta_{X};;#murad", 36, 0., 36.);
+    h_yPos = booker.book1D("Ypos", "Alignment fit #DeltaY;;#mum", 36, 0., 36.);
+    h_yRot = booker.book1D("Yrot", "Alignment fit #Delta#theta_{Y};;#murad", 36, 0., 36.);
+    h_zPos = booker.book1D("Zpos", "Alignment fit #DeltaZ;;#mum", 36, 0., 36.);
+    h_zRot = booker.book1D("Zrot", "Alignment fit #Delta#theta_{Z};;#murad", 36, 0., 36.);
+  } else {
+    booker.setCurrentFolder("AlCaReco/SiPixelAliHG/");
+
+    layerVec = {{"Layer1", 12},
```

Comment:
maybe all of this can be taken from ES instead of hardcoding?

https://github.com/cms-sw/cmssw/blob/a010308031c0537a386027d896611a1d738cd053/Geometry/TrackerGeometryBuilder/interface/PixelTopologyMap.h#L32-L35

Reply:
Sure, however from `getPXFModules` I get only 1 module for each disk. Is this expected? I could instead use `getPXFBlades` and multiply the result by 2.

Reply:
unfortunately as far as I can see from [here](https://github.com/cms-sw/cmssw/tree/6d2f66057131baacc2fcbdd203588c41c885b42c/Geometry/TrackerNumberingBuilder#phase-1-upgrade-detector-detid-schema) it is:

<ul dir="auto">
<li>Subdetector 2 (<code>DetId::subDetId() == PixelSubdetector::PixelEndcap</code>): Phase1 Pixel Forward</li>
</ul>


Name | start bit | hex mask | bit size | TrackerTopology method | Notes
-- | -- | -- | -- | -- | --
subdetector part | 23 | 0x3 | 2 | pxfSide(id) or side(id) | 1=FPIX- 2=FPIX+
not used | 22 | 0x1 | 1 |   |  
Blade | 12 | 0x3F | 6 | pxfBlade(id) | increasing phi and r: first inner ring blades and the outer ring blades
Panel | 10 | 0x3 | 2 | pxfPanel(id) | 1=forward 2=backward
Module | 2 | 0xFF | 8 | pxfModule(id) | always = 1

though admittedly that's not the way I thought it was supposed to work when I wrote the code.
I guess you can go ahead with the blade implementation while I pick up the slack...